controller/telemetry/api.go

```diff
@@ -5,24 +5,57 @@ import (
 	"time"
 
 	"github.com/reef-pi/reef-pi/controller/utils"
+	"github.com/reef-pi/reef-pi/controller/storage"
 )
 
+// TODO: translate these
+// TODO: this is a bit of a hack in that it means someone can't have their
+// 			 actual token/password be the string "<stored>", but that seems rare
+//			 enough a reasonable trade-off for a quick fix to unsaveable form bugs
+const PasswordStoredPlaceholder = "<stored>"
+const AdafruitIOTokenStoredPlaceholder = "<stored>"
+
 func (t *telemetry) GetConfig(w http.ResponseWriter, req *http.Request) {
 	fn := func(_ string) (interface{}, error) {
 		var c TelemetryConfig
 		if err := t.store.Get(t.bucket, DBKey, &c); err != nil {
 			return nil, err
 		}
-		c.AdafruitIO.Token = ""
-		c.Mailer.Password = ""
+		if (c.AdafruitIO.Token != "") {
```

Comment:
@richievos  i think we can get rid of the behavior of returning telemetry config with the password scrubbed completely.. and remove this special placeholder logic. After 6 years of use i find it more pain than a security feature